docs/source/type_narrowing.rst

```diff
@@ -83,6 +84,38 @@ We can also use ``assert`` to narrow types in the same context:
      reveal_type(x)  # Revealed type is "builtins.int"
      print(x + '!')  # Typechecks with `mypy`, but fails in runtime.
 
+
+Limitations of narrowing
+~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
Hope this is the right place. I wanted it to have a heading, to make it easy to refer people, but it's also a bit strange that it comes as a topic in the same hierarchy and right before "is_subclass" and "callable".

Reply:
Let's make it a heading and move it to the bottom of the page.